src/collected_output.rs

```diff
@@ -30,8 +30,8 @@ impl Waiter {
         });
         let mut context_clone = context.clone();
         let relay_stdout = config.relay_stdout;
-        let stdout_join_handle = thread::spawn(move || -> io::Result<Vec<u8>> {
-            let mut collected_stdout = Vec::new();
+        let stdout_join_handle = thread::spawn(move || -> io::Result<Option<Vec<u8>>> {
+            let mut collected_stdout = if relay_stdout { None } else { Some(Vec::new()) };
             let buffer = &mut [0; 256];
```

Comment:
This buffer seems a little small, although that's not based on anything objective, just a hunch. Maybe `4096` instead?

Reply:
Maybe, yes. That shouldn't be done in this PR though.